src/frontend/mame/luaengine.cpp

```diff
@@ -1786,21 +1819,53 @@ void lua_engine::initialize()
 	image_type.set("is_reset_on_load", sol::property(&device_image_interface::is_reset_on_load));
 	image_type.set("must_be_loaded", sol::property(&device_image_interface::must_be_loaded));
 
+/*  device_slot_interface library
+ *
+ * manager:machine().slots[slot_name]
+ *
+ * slot:fixed()
+ * slot:has_selectable_options()
+ *
+ * slot.options[] - get options table (k=name, v=device_slot_interface::slot_option)
+ */
+
+	auto slot_type = sol().registry().new_usertype<device_slot_interface>("new", sol::no_constructor);
+	slot_type.set("fixed", &device_slot_interface::fixed);
+	slot_type.set("has_selectable_options", &device_slot_interface::has_selectable_options);
+	slot_type.set("default_option", &device_slot_interface::default_option);
+	slot_type.set("options", sol::property([this](device_slot_interface &slot) {
+			sol::table option_table = sol().create_table();
+			for (auto &x : slot.option_list())
+				option_table[x.first] = &*x.second;
+			return option_table;
+		}));
+	sol().registry().set("slot", slot_type);
```

Comment:
Once again, documentation, syntax, spacing, methods vs properties.  Also, it’s better to make a lightweight wrapper for the options rather than walking it and building a Lua table every time.